crypto/src/lib.rs

```diff
@@ -159,21 +162,17 @@ pub trait Crypto {
     /// * `digest` - Digest of data to be signed.
     fn ecdsa_sign_with_alias(algs: AlgLen, digest: &Digest) -> Result<EcdsaSig, CryptoError>;
 
-    /// Sign `digest` with a derived key from the CDI.
+    /// Sign `digest` with a derived key-pair from the CDI and caller-supplied private key
     ///
     /// # Arguments
     ///
     /// * `algs` - Which length of algorithms to use.
-    /// * `cdi` - CDI from which to derive the signing key
-    /// * `label` - Caller-supplied label to use in asymmetric key derivation
-    /// * `info` - Caller-supplied info string to use in asymmetric key derivation
     /// * `digest` - Digest of data to be signed.
+    /// * `priv_key` - Caller-supplied private key to use in public key derivation
     fn ecdsa_sign_with_derived(
         algs: AlgLen,
-        cdi: &Self::Cdi,
-        label: &[u8],
-        info: &[u8],
         digest: &Digest,
+        priv_key: &EcdsaPriv,
```

Comment:
So the reason this takes a cdi instead of a private key is that in some integrations they actual private key will be held in hardware and firmware will never see the private key.

In the Caliptra implementation of this trait, I think the Cdi will just be the index of a KeyVault slot.

So at least for now I think it's better to rederive the key from the CDI whenever it is needed.